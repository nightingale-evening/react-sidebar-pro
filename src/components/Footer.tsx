import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

export interface FooterProps extends React.HTMLAttributes<HTMLHtmlElement> {
  height?: string;
}

const StyledFooter = styled.footer<FooterProps>`
  height: ${({ height }) => height};
  min-height: ${({ height }) => height};
`;

export const Footer: React.FC<FooterProps> = React.memo(
  ({ height = '64px', className, children, ...rest }) => {
    return (
      <StyledFooter
        data-testid="footer-test-id"
        className={classnames('footer', className)}
        height={height}
        {...rest}
      >
        {children}
      </StyledFooter>
    );
  },
);
