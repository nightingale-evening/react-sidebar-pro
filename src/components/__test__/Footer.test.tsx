import React from 'react';
import { customRender, screen } from '../../utils/testUtils';
import { Footer } from '../Footer';

describe('Footer', () => {
  it('basic snapshot ', () => {
    const { container } = customRender(<Footer>Footer</Footer>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should initialize Footer correctly', async () => {
    customRender(<Footer>Footer</Footer>);
    const footerElem = screen.getByText('Footer');

    expect(footerElem).toBeInTheDocument();
    expect(footerElem).toHaveClass('footer');
    expect(footerElem).toHaveStyle({
      height: '64px',
      'min-height': '64px',
    });
  });

  it('should set the height to 50px', async () => {
    customRender(<Footer height="50px">Footer</Footer>);
    const footerElem = screen.getByTestId('footer-test-id');

    expect(footerElem).toHaveStyle({
      height: '50px',
      'min-height': '50px',
    });
  });
});
