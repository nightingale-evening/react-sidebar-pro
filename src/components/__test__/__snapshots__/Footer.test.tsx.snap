// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer basic snapshot  1`] = `
<footer
  class="sc-bdvvtL cGdMXw footer"
  data-testid="footer-test-id"
  height="64px"
>
  Footer
</footer>
`;
